### PR TITLE
Monitor new version privatelinkscopes 2021-07-01-preview [PrivateLinkScopes]

### PR DESCRIPTION
**Pull request by arolshan**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by arolshan** [comment]
1005 - RemovedPath
-	Just aligned the paths to PascalCase.
1011 - AddingResponseCode
-	Added default response. 
1026 - TypeChanged
-	Added type property where required.

**Comment by filizt** [comment]
LGTM.

File: specification/monitor/resource-manager/readme.python.md [hunk 1]
```diff
@@ -129,6 +150,11 @@ Please also specify `--python-sdks-folder=<path to the root directory of your az
 ``` yaml $(tag) == 'package-2019-10-17-preview-only'
 namespace: $(python-base-namespace).v2019_10_17
 output-folder: $(python-sdks-folder)/$(python-base-folder)/v2019_10_17
+directive:
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/readme.python.md:153]: Not sure why remove `ResourceGroupNameParameter` here.

Comment (arolshan) [reply specification/monitor/resource-manager/readme.python.md:142]: moved it from specification/monitor/resource-manager/readme.md
it probably messed up the automatic sdk generation for the 2019 version. 

image

**Comment by arolshan** [comment]
❌ 1005 - RemovedPath | The new version is missing a path that was found in the old version. Was path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/privateLinkScopes' removed or restructured?Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L70:5
❌ 1005 - RemovedPath | The new version is missing a path that was found in the old version. Was path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/privateLinkScopes/{scopeName}' removed or restructured?Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L103:5
❌ 1005 - RemovedPath | The new version is missing a path that was found in the old version. Was path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/privateLinkScopeOperationStatuses/{asyncOperationId}' removed or restructured?Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L260:5
**---
This is just to align with the conventions. No a real breaking change.** 

❌ 1006 - RemovedDefinition | The new version is missing a definition that was found in the old version. Was 'PrivateEndpointProperty' removed or renamed?New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L846:3Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L745:3
❌ 1006 - RemovedDefinition | The new version is missing a definition that was found in the old version. Was 'PrivateLinkServiceConnectionStateProperty' removed or renamed?New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L846:3Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L745:3
❌ 1006 - RemovedDefinition | The new version is missing a definition that was found in the old version. Was 'PrivateEndpointConnection' removed or renamed?New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L846:3Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L745:3
**---
I aligned with common types of private links.** 


❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L373:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L418:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L467:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L526:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L577:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L620:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L669:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L734:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L785:11
❌ 1011 - AddingResponseCode | The new version adds a response code 'default'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L828:11
**---
This was to fix lintdiff. This actually a feature not a bug.** 


❌ 1029 - ReadonlyPropertyChanged | The read only property has changed from 'true' to 'false'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L850:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L890:9
❌ 1029 - ReadonlyPropertyChanged | The read only property has changed from 'false' to 'true'.New: common-types/resource-management/v2/privatelinks.json#L12:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L963:9
❌ 1029 - ReadonlyPropertyChanged | The read only property has changed from 'true' to 'false'.New: common-types/resource-management/v2/privatelinks.json#L68:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L985:9
❌ 1029 - ReadonlyPropertyChanged | The read only property has changed from 'true' to 'false'.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L1055:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1012:9
❌ 1029 - ReadonlyPropertyChanged | The read only property has changed from 'false' to 'true'.New: common-types/resource-management/v2/types.json#L289:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1102:9
❌ 1029 - ReadonlyPropertyChanged | The read only property has changed from 'false' to 'true'.New: common-types/resource-management/v2/types.json#L294:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1106:9
❌ 1033 - RemovedProperty | The new version is missing a property found in the old version. Was 'nextLink' renamed or removed?New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L1054:7Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1011:7
❌ 1033 - RemovedProperty | The new version is missing a property found in the old version. Was 'code' renamed or removed?New: common-types/resource-management/v2/types.json#L326:7Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1101:7
❌ 1033 - RemovedProperty | The new version is missing a property found in the old version. Was 'message' renamed or removed?New: common-types/resource-management/v2/types.json#L326:7Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1101:7
❌ 1034 - AddedRequiredProperty | The new version has new required property 'privateLinkServiceConnectionState' that was not found in the old version.New: Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json#L1039:9Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L1001:9
❌ 1034 - AddedRequiredProperty | The new version has new required property 'privateLinkServiceConnectionState' that was not found in the old version.New: common-types/resource-management/v2/privatelinks.json#L36:5Old: Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json#L940:5
**---
Everything here is to align with privatelink.json common types**

**Comment by arolshan** [comment]
Change PrivateEndpointConnection to AzureMonitorPrivateEndpointConnection
is this a breaking change?

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json [hunk 2]
```diff
@@ -346,6 +346,9 @@
         ],
         "description": "Gets the private link resources that need to be created for a Azure Monitor PrivateLinkScope.",
         "operationId": "PrivateLinkResources_ListByPrivateLinkScope",
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json:350]: `"nextLinkName": null` would be fine.

**Comment by arolshan** [comment]
Done.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json [hunk 3]
```diff
@@ -541,13 +614,16 @@
...
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
             }
           }
         },
-        "x-ms-pageable": {
-          "nextLinkName": "nextLink"
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json]: Why is this removed?

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json [hunk 4]
```diff
@@ -541,13 +614,16 @@
           "200": {
             "description": "Successfully retrieved private endpoint connections.",
             "schema": {
-              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+              "$ref": "#/definitions/ExtendedPrivateEndpointConnectionListResult"
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json:617]: Why changed from `PrivateEndpointConnectionListResult` to  `ExtendedPrivateEndpointConnectionListResult`

Change type name will be breaking changes and would require PR to go through breaking changes review board.

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json (hunk 3 above)
Comment (arolshan) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json]: can u point to the API?

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json (hunk 4 above)
Comment (arolshan) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json:617]: we are extending the common PrivateEndpointConnection type and adding 2 properties. 
Keeping the old PrivateEndpointConnection name will cause confusion with common type here:
"../../../../../common-types/resource-management/v2/privatelinks.json"

Do you suggest just cloning the definition in the common type and thus keep the old name?

Comment (weidongxu-microsoft) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2021-07-01-preview/privateLinkScopes_API.json:617]: Sorry, I am not able to suggestion what is the best solution, as there is too much changes in the PR (and many of them involves renaming model, or replace some model with another model from common-types json), and I am really not able to tell which change causes breaks, which does not.

**Comment by weidongxu-microsoft** [comment]
@msyyc @ArcturusZhang for awareness of breaking changes.

**Comment by arolshan** [comment]
Please compare with 2019-10-17-Preview to validate no breaking changes.

**Comment by arolshan** [comment]
Please don't merge this PR without consent.
